bcipy/task/paradigm/matrix/calibration.py

```diff
@@ -0,0 +1,269 @@
+from bcipy.display.paradigm.matrix.mode.calibration import CalibrationDisplay
+from bcipy.helpers.clock import Clock
+from bcipy.helpers.stimuli import (StimuliOrder, calibration_inquiry_generator,
+                                   get_task_info)
+from bcipy.helpers.task import (alphabet, get_user_input, pause_calibration,
+                                trial_complete_message)
+from bcipy.helpers.triggers import TriggerHandler, TriggerType, Trigger, FlushFrequency, convert_timing_triggers
+
+from bcipy.task import Task
+
+
+class MatrixCalibrationTask(Task):
+    """Matrix Calibration Task.
+
+    Calibration task performs an Matrix stimulus inquiry
+        to elicit an ERP. Parameters will change how many stimuli
+        and for how long they present. Parameters also change
+        color and text / image inputs.
+
+    A task begins setting up variables --> initializing eeg -->
+        awaiting user input to start -->
+        setting up stimuli --> presenting inquiries -->
+        saving data
+
+    PARAMETERS:
+    ----------
+    win (PsychoPy Display Object)
+    daq (Data Acquisition Object)
+    parameters (Dictionary)
+    file_save (String)
+    """
+
+    def __init__(self, win, daq, parameters, file_save):
+        super(MatrixCalibrationTask, self).__init__()
+
+        # TODO fix stimuli size for the matrix keyboard to present well
+        # Update stimuli generation for matrix
+        # fix triggers and create a write method to ensure they write correctly using
+
+        self.window = win
+        self.frame_rate = self.window.getActualFrameRate()
+        self.parameters = parameters
+        self.daq = daq
+        self.static_clock = core.StaticPeriod(screenHz=self.frame_rate)
+        self.experiment_clock = Clock()
+        self.buffer_val = parameters['task_buffer_len']
+        self.alp = alphabet(parameters)
+        self.matrix = init_calibration_display_task(
+            self.parameters, self.window,
+            self.static_clock, self.experiment_clock)
+        self.file_save = file_save
+        self.trigger_handler = TriggerHandler(
+            self.file_save,
+            parameters['trigger_file_name'],
+            FlushFrequency.EVERY)
+
+        self.wait_screen_message = parameters['wait_screen_message']
+        self.wait_screen_message_color = parameters[
+            'wait_screen_message_color']
+
+        self.stim_number = parameters['stim_number']
+        self.stim_length = parameters['stim_length']
+        self.stim_order = StimuliOrder(parameters['stim_order'])
+
+        self.timing = [parameters['time_flash']]
+
+        self.color = [parameters['stim_color']]
+
+        self.task_info_color = parameters['task_color']
+
+        self.stimuli_height = parameters['stim_height']
+
+        self.is_txt_stim = parameters['is_txt_stim']
+        self.eeg_buffer = parameters['eeg_buffer_len']
+
+        self.enable_breaks = parameters['enable_breaks']
```

Comment:
We are already setting `parameters` as an instance variable. Rather than having a bunch of additional instance variables that just wrap these values, my preference is just to use `self.parameter[...]`. If the value is updated by other methods or is a computed value it should definitely stay an instance variable, though.

Reply:
since there is no standard, I think this should go through right now as is. Let's surface this as a style discussion next we meet as a group. I can see drawbacks to both, but don't dislike how we've changed copy phrase.